addons/sale_stock/static/src/js/qty_at_date_widget.js

```diff
@@ -45,15 +45,40 @@ var QtyAtDateWidget = Widget.extend({
             this._setPopOver();
         }
     },
-    //--------------------------------------------------------------------------
-    // Private
-    //--------------------------------------------------------------------------
     /**
-     * Set a bootstrap popover on the current QtyAtDate widget that display available
-     * quantity.
+     * Redirect to the product graph view.
+     *
+     * @private
+     * @param {MouseEvent} event
+     * @returns {Promise} action loaded
      */
-    _setPopOver: function () {
-        var self = this;
+    async _openForecast(ev) {
+        ev.stopPropagation();
+        const action = await data_manager.load_action('stock.report_stock_quantity_action_product');
+            // Change action context to choose a specific date and product(s)
+            // As grid_anchor is set to now() by default in the data, we need
+            // to load the action first, change the context then launch it via do_action
+            // additional_context cannot replace a context value, only add new
+            //
+            // in case of kit product, the forecast view show the kit's components
+        const res = await this._rpc({
```

Comment:
careful about performance when dealing with multiple promise-based async methods
you code will do this:
- ask for the action to load, await the result and store it in the `const action` variable
- ask for an rpc call, await the result and store it in the `const res` variable
=> there will be 2 waiting periods, the rpc will only be started once the action has been fully loaded

For performance reason, you should always try to _group_ your async calls in a single promise using  `Promise.all`, so that they are done in parallel rather than sequentially. You can use [destructuring assignments](https://www.freecodecamp.org/news/array-destructuring-in-es6-30e398f21d10/) to make it easy to read:
```js
const [action, res] = await Promise.all([
    data_manager.load_action('stock.report_stock_quantity_action_product'),
    this._rpc({
        model: 'product.product',
        method: 'get_components',
        args: [[this.data.product_id.data.id]]
    })
)];
```

See the [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all) entry for examples and details.

Reply:
@dbo-odoo Done. Thanks for your brief explanation. :)